src/tmpfiles/tmpfiles.c

```diff
@@ -1528,8 +1528,7 @@ static int create_directory_or_subvolume(Item *i, const char *path) {
                         RUN_WITH_UMASK((~i->mode) & 0777)
                                 r = btrfs_subvol_make(i->path);
                 }
-        } else
-                r = 0;
+        }
```

Comment:
dropping this looks like something that will trip up gcc sooner or later, as it tends to be too dumb to realize that r is always initialized in the r == -ENOTTY check in the if line immediately following...

Moreover, I'd prefer if we didn't write this code so fragile, as the order of the checks in the following if line suddenly matters, which is very much not obvious